src/debugger/forkedAppWorker.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as path from "path";
+import * as child_process from "child_process";
+import {ScriptImporter}  from "./scriptImporter";
+import {FileSystem} from "../common/node/fileSystem";
+
+import { Log } from "../common/log/log";
+import { LogLevel } from "../common/log/logHelper";
+import { ErrorHelper } from "../common/error/errorHelper";
+import { IDebuggeeWorker, RNAppMessage } from "./appWorker";
+
+// tslint:disable-next-line:align
+const WORKER_BOOTSTRAP = `
+// Hacks to avoid accessing unitialized variables
+var onmessage=null, self=global;
+// Avoid Node's GLOBAL deprecation warning
+global.GLOBAL=global;
+process.on("message", function(message){
+    if (onmessage) onmessage(message);
+});
+postMessage = function(message){
+    process.send(message);
+};
+importScripts = function(scriptUrl){
+    var scriptCode = require("fs").readFileSync(scriptUrl, "utf8");
+    require("vm").runInThisContext(scriptCode, { filename: scriptUrl });
+};`;
```

Comment:
Something else that I would recommend is adding a global `__debug__` object with a `require` function which caches the node `require`. This matches what we currently expose in the original sandboxing method, and allows things such as realm to provide shims for native functionality when necessary.